### DISCUSSION TITLE
How to move forward

### DISCUSSION THREAD

**Discussion by objecttothis** [opensourcepos/opensourcepos#3680] [Team Posts]
Hey @opensourcepos/org @odiea posted in the gitter dev room and I thought I would post this here to start the dialog.  I know that @jekkos and @daN4cat have not had a lot of time to give to this software in the recent past.  I want to balance that statement with my need to continue development on changes that we need for the website and bookstore and that I think the community could benefit from.  The problem lies in the things that depend on @jekkos and @daN4cat for.  I don't own the code, so I don't feel liberty to just merge things into the master.  Our current setup doesn't allow it anyway (and that's as it should be).  The problem is that I have two major pull requests that can't move forward until they get signed off on.  The optimizations PR and the event generator.  The latter, in conversation with @jekkos we have agreed that there is likely a better implementation that is more robust, but that means I'm waiting on both PR's and I don't want to implement more changes until at least the optimizations/CSV update import PR has been merged.

On one side I don't want to put pressure on people, but on the other side I need to keep moving.  There are three reasons that I prefer not to fork.  One is that I feel that enhancements and bug fixes that I contribute can benefit the whole community.  Two is that I don't want to maintain future changes that get made to opensourcepos into a fork.  Finally, there are a number of systems that go into making ospos work (opensourcepos.org, weblate, TravisCI, etc).  I would need to learn how to build the software and connect it to a new weblate which is more work that I prefer not to employ at this time.

All that said, what do we do?  @jekkos @daN4cat thoughts?  Again, I have no goal to ruffle feathers or put pressure on people.  Simply put, how do we move forward?

**Comment by jekkos** [comment]
Hi @objecttothis I understand the issue here and would like to help and unblock you. Personally I do have a bit less time to invest in the project, partly because my parent's business is looking for a buyer. Once they have someone I  will not be needed anymore to support their version of ospos. Let's see how that goes.

On the other hand I still enjoy maintaining the infrastructure and helping out the users and community here.

So I don't see myself adding any new bigger features in the near term. On the other hand I'm still interested in getting the event improvements worked out, but I will have to find some time to do that. It's good to hear that you have things working on your side already, so that I'm not blocking you

Beside this I think we are mostly good to have a 3.3.3. After this I'll try to review your pull requests again so we can get things moving for 3.4. 

I also want to say that I already considered to give you, steve and odie admin rights as well, so you guys can jump in when needed. I think that this will be better for all in the long term. Ofc with great power comes great responsibility.

**Comment by objecttothis** [comment]
@jekkos, thank you for your response.  Once we get these two PR's figured out, I think the main thing that can act as a bottle neck is that I want to respect that you are the code owner and therefore design decisions and roadmap decisions are something that I don't feel comfortable with making nor the authority to do so.  Even though the Optimization and Event Generator PRs are running on my production environment, I can't make further changes to submit a PR for without risking a good amount of rebasing issues.  In that respect I am bottlenecked.  That said, please attend to your parent's business since that is more important for you and your family at the moment than this.  I have some other projects to keep me busy.

**Comment by jekkos** [comment]
Ok I have assigned you admin rights.. Same for @SteveIreland and @odiea
